### PR TITLE
fix(nuxt): do not include `<NuxtWelcome>` when building

### LINKED ISSUE #29838: Unable to load module.ts as the entry point of the local module.
### Environment

- Operating System: Linux
- Node Version:     v18.20.3
- Nuxt Version:     3.14.159
- CLI Version:      3.15.0
- Nitro Version:    2.10.3
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      default
- Runtime Modules:  ./modules/my-module
- Build Modules:    -

### Reproduction

[Working reproduction (3.13.0)](https://stackblitz.com/edit/nuxt-starter-8fd3te?file=package.json)
[Not working reproduction (3.14.159)](https://stackblitz.com/edit/nuxt-starter-gy1kco?file=package.json)

### Describe the bug

When the entry point of the local module is module.ts, it works in version 3.13.0 but does not work in 3.14.159. However, if the entry point is index.ts, it works in 3.14.159.

### Additional context

_No response_

### Logs

❯ npm install && npm run dev

> postinstall
> nuxt prepare

 ERROR  Nuxt module should be a function: ./modules/my-module                                                 오후 3:41:12

  at loadNuxtModuleInstance (node_modules/@nuxt/kit/dist/index.mjs:2549:11)
  at async Module.installModule (node_modules/@nuxt/kit/dist/index.mjs:2478:47)
  at async initNuxt (node_modules/nuxt/dist/index.mjs:4686:5)
  at async loadNuxt (node_modules/nuxt/dist/index.mjs:4856:5)
  at async loadNuxt (node_modules/@nuxt/kit/dist/index.mjs:2716:19)
  at async Object.run (node_modules/nuxi/dist/chunks/prepare.mjs:76:18)
  at async runCommand$1 (node_modules/nuxi/dist/shared/nuxi.bd0a2fa0.mjs:1680:16)
  at async runCommand$1 (node_modules/nuxi/dist/shared/nuxi.bd0a2fa0.mjs:1671:11)
  at async runMain$1 (node_modules/nuxi/dist/shared/nuxi.bd0a2fa0.mjs:1809:7)

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/29838

### 📚 Description

This should avoid bundling `<NuxtWelcome>` - though we can also investigate why it's being included in the main bundle rather than as an async chunk....

## Summary by CodeRabbit

- **New Features**
	- Introduced conditional logic for the `NuxtWelcome` component, enhancing component management based on user settings.
	- Maintained the addition of components like `NuxtLayout`, `NuxtErrorBoundary`, and others without modifications.

- **Documentation**
	- Added comments for clarity on handling experimental features and component registration.

### CODE REVIEW

File: packages/nuxt/src/core/nuxt.ts [hunk 1]
```diff
@@ -408,11 +408,14 @@ async function initNuxt (nuxt: Nuxt) {
   }
 
   // Add <NuxtWelcome>
-  addComponent({
-    name: 'NuxtWelcome',
-    priority: 10, // built-in that we do not expect the user to override
-    filePath: resolve(nuxt.options.appDir, 'components/welcome'),
-  })
+  const islandsConfig = nuxt.options.experimental.componentIslands
+  if (nuxt.options.dev || !islandsConfig || islandsConfig === true || typeof islandsConfig === 'string' || islandsConfig.selectiveClient !== 'deep') {
```
Comment (huang-julien) [inline packages/nuxt/src/core/nuxt.ts:412]: Can we make this shorter  ?

**Comment by huang-julien** [comment]
There's quite a few things that need to be fixed with how we chunk when `selectiveClient` is enabled. But first i need to finish the rework of island

File: packages/nuxt/src/core/nuxt.ts (hunk 1 above)
Comment (danielroe) [reply packages/nuxt/src/core/nuxt.ts:412]: ideas welcome! - but we also have to type guard given that `componentIslands` can be a string, boolean or object...

**Comment by danielroe** [comment]
happy to skip this for now if you want as it's a bit of an edge case.

File: packages/nuxt/src/core/nuxt.ts (hunk 1 above)
Comment (huang-julien) [reply packages/nuxt/src/core/nuxt.ts:412]:
```suggestion
  if (!(typeof islandConfig === 'object' && islandsConfig.selectiveClient === 'deep') {
```

Comment (huang-julien) [reply packages/nuxt/src/core/nuxt.ts:412]: like this ?

**Comment by huang-julien** [comment]
We can merge it, but i'm pretty sure there's many other components that are probably also included in the bundle

File: packages/nuxt/src/core/nuxt.ts (hunk 1 above)
Comment (danielroe) [reply packages/nuxt/src/core/nuxt.ts:412]: perfect!

**Pull request merged**